cas-server-support-saml-idp/src/main/java/org/jasig/cas/support/saml/SamlIdPServletContextListener.java

```diff
@@ -0,0 +1,31 @@
+package org.jasig.cas.support.saml;
+
+import org.jasig.cas.web.AbstractServletContextInitializer;
+import org.jasig.cas.web.support.WebUtils;
+import org.springframework.stereotype.Component;
+
+import javax.servlet.ServletContextEvent;
+import javax.servlet.annotation.WebListener;
+
+/**
+ * Initializes the CAS root servlet context to make sure
+ * SAML IdP can be activated by the main CAS servlet.
+ * @author Misagh Moayyed
+ * @since 4.2
+ */
+@WebListener
+@Component("samlIdPServletContextListener")
+public class SamlIdPServletContextListener extends AbstractServletContextInitializer {
+
+    @Override
+    public void initializeServletContext(final ServletContextEvent event) {
+        if (WebUtils.isCasServletInitializing(event)) {
+            addEndpointMappingToCasServlet(event, SamlIdPConstants.ENDPOINT_IDP_METADATA);
```

Comment:
Controllers are, yes but this is not adding a controller. This is simply adding a mapping for that endpoint in the web.xml file at runtime. This is needed, because we map everything explicitly, rather than mapping /* and letting components handle things. The latter approach is easier, but I also think explicitness in declaring what endpoint is supported at the container level reduces "the attack surface".